lib/Support/regmalloc.cpp

```diff
@@ -0,0 +1,36 @@
+//===-- regmalloc.cpp - Memory allocation for regex implementation --------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+// Support operator new/delete overriding for regex memory allocations.
+//===----------------------------------------------------------------------===//
+
+#include "regutils.h"
+#include <algorithm>
+#include <new>
+#include <cstring>
+
+extern "C" {
+  void *re_malloc(size_t size) {
```

Comment:
I read this as re-malloc (i.e., some version of realloc)... perhaps regex_malloc?

Reply:
sure, I was trying to match some function names of the C regex code, but even they are not consistent internally so I can do that.